### PR TITLE
MINOR: improve test stability for Streams broker-compatibility test

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
https://jenkins.confluent.io/job/system-test-kafka-branch-builder-2/253/

**Comment by mjsax** [comment]
Call for review @enothereska @dguy

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py
Comment (edenhill) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: nit:
The error text can be misleading in the case where more than one message was received, the original assert seem to have covered this case (in its error message).
Maybe change the wait to ' total_consumed() > 0' and then keeping the original assert for actual verification and error message?

Comment (mjsax) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: Well. The Streams app writes exactly one message -- thus, it seems to be fair to expect exactly one. No?

I am wondering, how it could happen that it fails with more than one? It should either be zero if something went wrong or one.

File: streams/src/test/java/org/apache/kafka/streams/tests/BrokerCompatibilityTest.java
Comment (enothereska) [inline streams/src/test/java/org/apache/kafka/streams/tests/BrokerCompatibilityTest.java]: I think this is already the default.

**Comment by enothereska** [comment]
LGTM

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py
Comment (edenhill) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: Right, I was referring to the difference to the old assert and that the new check and error message contains less troubleshooting information than the old one, but I understand that it might seem irrelevant given that the producer part can be trusted, thus the "nit" :)

**Comment by mjsax** [comment]
Updated.

**Comment by mjsax** [comment]
Call for second review @dguy @edenhill

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py
Comment (edenhill) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:74]: This test went from checking for exactly one message, to checking at least one message.
If you think that is fine (I dont have the full scope so I dont know) then LGTM! :)

Comment (mjsax) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:74]: Yes. The test is just to make sure, Streams can connect to brokers of older versions -- the Streams App is super basic just reading a record for an input topic and write it to the output topic.

**Review by dguy — APPROVED** [review]
LGTM

**Comment by mjsax** [comment]
Call for final review and merging @guozhangwang

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
LGTM and merged to trunk.

**Comment by mjsax** [comment]
@guozhangwang This could go to `0.10.2`, too. WDYT?

**Pull request closed without merging**
